nullaway/src/main/java/com/uber/nullaway/autofix/AutoFixConfig.java

```diff
@@ -0,0 +1,351 @@
+    JSONObject jsonObject;
+    try {
+      Object obj =
+          new JSONParser().parse(Files.newBufferedReader(filePath, Charset.defaultCharset()));
+      jsonObject = (JSONObject) obj;
+    } catch (Exception e) {
+      throw new RuntimeException("Error in reading/parsing config at path: " + filePath);
+    }
+    MAKE_METHOD_TREE_INHERITANCE_ENABLED =
+        getValueFromKey(jsonObject, "MAKE_METHOD_INHERITANCE_TREE", Boolean.class).orElse(false)
+            && autofixEnabled;
+    MAKE_CALL_GRAPH_ENABLED =
+        getValueFromKey(jsonObject, "MAKE_CALL_GRAPH", Boolean.class).orElse(false)
+            && autofixEnabled;
+    MAKE_FIELD_GRAPH_ENABLED =
+        getValueFromKey(jsonObject, "MAKE_FIELD_GRAPH", Boolean.class).orElse(false)
+            && autofixEnabled;
+    SUGGEST_ENABLED =
+        getValueFromKey(jsonObject, "SUGGEST:ACTIVE", Boolean.class).orElse(false)
+            && autofixEnabled;
+    SUGGEST_DEEP =
+        getValueFromKey(jsonObject, "SUGGEST:DEEP", Boolean.class).orElse(false) && SUGGEST_ENABLED;
+    PARAM_TEST_ENABLED =
+        getValueFromKey(jsonObject, "METHOD_PARAM_TEST:ACTIVE", Boolean.class).orElse(false)
+            && autofixEnabled;
+    PARAM_INDEX =
+        Math.toIntExact(
+            getValueFromKey(jsonObject, "METHOD_PARAM_TEST:INDEX", Long.class)
+                .orElse(Long.MAX_VALUE));
+    LOG_ERROR_ENABLED =
+        getValueFromKey(jsonObject, "LOG_ERROR:ACTIVE", Boolean.class).orElse(false)
+            && autofixEnabled;
+    LOG_ERROR_DEEP =
+        getValueFromKey(jsonObject, "LOG_ERROR:DEEP", Boolean.class).orElse(false)
+            && autofixEnabled;
+    INHERITANCE_CHECK_DISABLED =
+        (getValueFromKey(jsonObject, "INHERITANCE_CHECK_DISABLED", Boolean.class).orElse(false)
+                && autofixEnabled)
+            || PARAM_TEST_ENABLED;
+    OPTIMIZED =
+        getValueFromKey(jsonObject, "OPTIMIZED", Boolean.class).orElse(false) && autofixEnabled;
+    String nullableAnnot =
+        getValueFromKey(jsonObject, "ANNOTATION:NULLABLE", String.class)
+            .orElse("javax.annotation.Nullable");
+    String nonnullAnnot =
+        getValueFromKey(jsonObject, "ANNOTATION:NONNULL", String.class)
+            .orElse("javax.annotation.Nonnull");
+    this.ANNOTATION_FACTORY = new AnnotationFactory(nullableAnnot, nonnullAnnot);
+    String WORK_LIST_VALUE = getValueFromKey(jsonObject, "WORK_LIST", String.class).orElse("*");
+    if (!WORK_LIST_VALUE.equals("*")) {
+      this.WORK_LIST = new HashSet<>(Arrays.asList(WORK_LIST_VALUE.split(",")));
+    } else {
+      this.WORK_LIST = Collections.singleton("*");
+    }
+    VIRTUAL_ANNOT_ENABLED =
+        getValueFromKey(jsonObject, "VIRTUAL:ACTIVE", Boolean.class).orElse(false)
+            && autofixEnabled;
+    VIRTUAL_ANNOT_PATH = getValueFromKey(jsonObject, "VIRTUAL:PATH", String.class).orElse("");
+    Writer.reset(this);
+  }
+
+  public boolean hasNullableAnnotation(Symbol symbol) {
+    return false;
```

Comment:
?

I think these methods (this one and the very next) imply dead code in the auto-fix logic. In particular, since these will statically always return `false`, this logic can be removed in Nullness.java:

```
|| (config.getAutoFixConfig().VIRTUAL_ANNOT_ENABLED
            && config.getAutoFixConfig().hasNullableAnnotation(symbol))
```

which in turn makes it worthwhile to check if `VIRTUAL_ANNOT_ENABLED` is still being used.

Reply:
They are not getting used anymore, and both are deleted now.